index.js

```diff
@@ -8,51 +8,122 @@ const RGBot = {
+  }
+}
+
+let charType = Math.round(Math.random() * 1000000) % 4;
+
+export function configureBot(characterType) {
+  console.log(`Unity bot configureBot function called, charType: ${charType} - characterType: ${characterType}`)
+  charType = CharInfo.type.indexOf(characterType)
 }
 
 
 let CURRENT_ABILITY = 0;
 
-export async function runTurn(tickInfo, mostRecentMatchInfo, actionQueue) {
+let lastEnemyId = -1;
+
+export async function runTurn(playerId, tickInfo, mostRecentMatchInfo, actionQueue) {
+
+  //console.log(`Running 'runTurn' with playerId: ${playerId}, tickInfo: ${JSON.stringify(tickInfo)}`)
+
+  // select 1 ability per tick
+  selectAbility(playerId, tickInfo, mostRecentMatchInfo, actionQueue);
+
 
-  console.log(tickInfo)
+  //TODO: Add script sensors to the door and button so that a bot can walk to a button if door not open
+}
 
-  // Collect some initial information that is needed
-  const enemies = BossRoomBot.getEnemies(tickInfo);
-  console.log(`Found ${enemies.length} enemies!`);
+function selectAbility(playerId, tickInfo, mostRecentMatchInfo, actionQueue) {
+  const myPlayer = BossRoomBot.getAlly(tickInfo, playerId);
+  //console.log(`My player is at position: ${JSON.stringify(myPlayer.position)}`)
 
-  // Some abilities require an enemy id and position
-  const randomEnemy = enemies[Math.floor(Math.random() * enemies.length)];
-  const ability = BossRoomBot.abilities[CURRENT_ABILITY % BossRoomBot.abilities.length];
+  // Some abilities require an enemy/ally id and position
+  const abilities = CharInfo.abilities[charType];
+  console.log(`Considering abilities ${JSON.stringify(abilities)}`);
+  const abilityIndex = CURRENT_ABILITY % abilities.length;
+  const ability = abilities[abilityIndex];
   console.log(`Trying out ability ${ability}`);
-  BossRoomBot.startAbility(ability, randomEnemy.position, randomEnemy.id, actionQueue);
+  const targetType = CharInfo.abilityTargets[charType][abilityIndex]
+  if ( targetType == 1) {
+    const enemies = BossRoomBot.getEnemies(tickInfo);
+    console.log(`Found ${enemies.length} enemies!`);
+    let randomEnemy = BossRoomBot.getEnemy(tickInfo, lastEnemyId);
+    if (!randomEnemy) {
+      randomEnemy = BossRoomBot.nearestEnemy(tickInfo, myPlayer.position);
+    }
+    if (randomEnemy) {
+      lastEnemyId = randomEnemy.id;
+      BossRoomBot.startAbility(ability, randomEnemy.position, randomEnemy.id, actionQueue);
+    } else {
+      lastEnemyId = -1;
+    }
+  } else {
+    const allies = BossRoomBot.getAllies(tickInfo);
+    console.log(`Found ${allies.length} allies!`);
+    let randomAlly;
+    if (targetType == -1 ){
+      randomAlly = null;
+    } else {
+      randomAlly = allies[Math.floor(Math.random() * allies.length)];
+    }
+    BossRoomBot.startAbility(ability, randomAlly ? randomAlly.position : null, randomAlly ? randomAlly.id : null, actionQueue);
+  }
   CURRENT_ABILITY++;
+}
 
-}
\ No newline at end of file
+export function getCharacterType() {
```

Comment:
I still need to go to the RegressionGames PR, but I am guessing this is called somewhere in that code? Might it make sense to make this a more general `provideGameConfig` function that returns a JSON?

Reply:
once we have a second config, i agree we can fixit on a good pattern

Reply:
Alrighty, I suppose it's fine to wait for now :P